buildSrc/src/main/java/io/dronefleet/mavlink/generator/MessageGenerator.java

```diff
@@ -222,7 +235,23 @@ public TypeSpec generate() {
                         .build())
                 .addMethod(generateEquals())
                 .addMethod(generateHashCode())
-                .addMethod(generateToString())
-                .build();
+                .addMethod(generateToString());
+
+        boolean hasTransmissionId = fields.stream()
+                .anyMatch(field -> "tid".equals(field.getName()) && "uint8_t".equals(field.getType()));
+        if (hasTransmissionId){
+            builder.addSuperinterface(ParameterizedTypeName.get(HAS_TRANSMISSION_ID, className));
+            builder.addMethod(MethodSpec
+                    .methodBuilder("withTransmissionId")
```

Comment:
What's the use of `withTransmissionId`? When are we going to copy a message and just change the tid?

Reply:
The tids will managed at a low level (in MavlinkConnectionWrapper), so it needs the ability to change it in `send()`.

Reply:
Keep in mind it means allocating every message twice.

Reply:
Not too worried, we're not sending that many.